src/command/supergraph/compose/do_compose.rs

```diff
@@ -119,6 +154,143 @@ impl Compose {
+            .opts
+            .clone()
+            .supergraph_config_source()
+            .clone()
+            .supergraph_yaml
+            .and_then(|file| match file {
+                FileDescriptorType::File(file) => {
+                    let mut current_dir =
+                        current_dir().expect("Unable to get current directory path");
+
+                    current_dir.push(file);
+                    let path = Utf8PathBuf::from_path_buf(current_dir).unwrap();
+                    let parent = path.parent().unwrap().to_path_buf();
+                    Some(parent)
+                }
+                FileDescriptorType::Stdin => None,
+            });
+
+        // Get a FullyResolvedSupergraphConfig from first loading in any remote subgraphs and then
+        // a local supergraph config (if present) and then combining them into a fully resolved
+        // supergraph config
+        let resolver = SupergraphConfigResolver::default()
+            .load_remote_subgraphs(
+                &studio_client,
+                self.opts.supergraph_config_source.graph_ref.as_ref(),
+            )
+            .await?
+            .load_from_file_descriptor(
+                &mut stdin,
+                self.opts.supergraph_config_source.supergraph_yaml.as_ref(),
+            )?
+            .fully_resolve_subgraphs(&client_config, &studio_client, supergraph_root.as_ref())
+            .await?;
+
+        // We convert the FullyResolvedSupergraphConfig into a Supergraph because it makes using
+        // Serde easier (said differently: we're using the Federation-rs types here for
+        // compatability with Federation-rs tooling later on when we use their supergraph binary to
+        // actually run composition)
+        let supergraph_config: SupergraphConfig = resolver.clone().try_into()?;
+
+        // Convert the FullyResolvedSupergraphConfig to yaml before we save it
+        let supergraph_config_yaml = serde_yaml::to_string(&supergraph_config)?;
+
+        // We're going to save to a temporary place because we don't actually need the supergraph
+        // config to stick around; we only need it on disk to point the supergraph binary at
+        let supergraph_config_filepath =
+            Utf8PathBuf::from_path_buf(tempdir()?.path().join("supergraph.yaml"))
+                .expect("Unable to parse path");
+
+        // Write the supergraph config to disk
+        let _ = write_file
+            .write_file(
+                &supergraph_config_filepath,
+                supergraph_config_yaml.as_bytes(),
+            )
+            .await?;
+
+        // Use the CLI option for federation over the one we can read off of the supergraph config
+        // (but default to the one we can read off the supergraph config)
+        let fed_version = if let Some(fed_version) = self.opts.federation_version.as_ref() {
+            fed_version
+        } else {
+            resolver.federation_version()
+        };
```

Comment:
You might be able to simplify this:
```suggestion
        let fed_version = self
                .opts
                .federation_version
                .as_ref()
                .unwrap_or(resolver.federation_version());
```